### PR TITLE
APP-9226 fix:Links adjacent to italics formatting breaking

### PR DESCRIPTION
**Pull request by achitJ**
### Summary
- Fixed inconsistent `target` attribute encoding in link generation to prevent markdown processing interference
- Added comprehensive test coverage for links adjacent to markdown formatting
- Improved code documentation and test code style consistency

#### 🐛 Bug Fix: Target Attribute Encoding
**Problem:** Links were using `target="_blank"` which could potentially interfere with markdown processing when underscores in HTML attributes are adjacent to markdown delimiters.

**Solution:** Changed both HTTP/HTTPS/FTP links and mailto links to use `target="&#95;blank"` (HTML entity-encoded underscore) for consistency and to prevent the markdown processor from matching underscores in HTML attributes with markdown delimiters.

**Files Changed:**
- `src/index.js` - Updated `target` attribute in link and mailto regex replacements
- Enhanced code comment explaining the rationale for using `&#95;` encoding

#### ✅ Test Improvements
Added 15 new comprehensive tests in `test/markdownTest.js` covering edge cases for links adjacent to markdown formatting:

**New Test Suite: "links adjacent to markdown formatting"**
- Links before/after/between italic text (`_text_`)
- Links before/after bold text (`*text*`)
- Links before/after strikethrough (`~text~`)
- Links before/after inline code (`` `code` ``)
- Links with underscores in URLs next to italic formatting
- Links with markdown characters in URLs (asterisks, underscores) adjacent to actual markdown
- Links with labels adjacent to markdown
- Multiple markdown formats surrounding links
- Edge case: No space between link and markdown delimiter

**Additional Improvements:**
- Updated existing test expectations to match the new `&#95;blank` encoding
- Improved code style consistency (quote usage in tests)
- Fixed minor linting issues (semicolons, quote consistency)

### Test Results
```
✓ 105 passing (138ms)
```
- **Previous:** 90 tests
- **New:** 105 tests (+15 tests)
- **All tests passing** ✅

Made sure the changes are working fine on local by linking the changes by:
- Outgoing emails
- UI story detail view
- Forward Sync messages to Zendesk

### Breaking Changes
None - This is a minor HTML output change that maintains the same functional behavior (links still open in new tabs with proper security attributes).

---
## Summary by cubic
Fixes links breaking when placed next to italic Markdown by encoding the underscore in target attributes, preventing Markdown collisions (APP-9226). Adds edge-case tests; link behavior remains unchanged.

- Bug Fixes
  - Use target="&#95;blank" for http/https/ftp and mailto links to avoid underscore matching with markdown.
  - Preserve rel="noopener noreferrer"; links still open in new tabs.
  - Add 15 tests for links adjacent to italics/bold/code/strikethrough and URLs with markdown chars; update expectations accordingly.

<sup>Written for commit a2b14152d27acfab6149c238b74358a8b2bbd4cc. Summary will update automatically on new commits.</sup>

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -503,7 +506,7 @@ const escapeForSlack = (text, options = {}) => {
         (match) =>
           `<a href="mailto:${
             match.mailTo
```
[the remark this answers is not in the document]
Comment (achitJ) [reply src/index.js:508]: This was existing behaviour - not changing much in this

File: src/index.js [hunk 2]
```diff
@@ -503,7 +506,7 @@ const escapeForSlack = (text, options = {}) => {
...
-          }" target="_blank" rel="noopener noreferrer">${
+          }" target="&#95;blank" rel="noopener noreferrer">${
```
[the remark this answers is not in the document]
Comment (achitJ) [reply src/index.js:509]: THis is fine since we are using this internally only

File: src/index.js [hunk 3]
```diff
@@ -482,15 +482,18 @@ const escapeForSlack = (text, options = {}) => {
...
+   * We use &#95; instead of _ in HTML attributes to prevent markdown processor from
+   * matching them with markdown delimiters
   */
   const textWithEncodedLink = XRegExp.replace(text || '',
     linkRegExp,
     (match) => {
       const encodedLink = encodeSlackMrkdwnCharactersInLinks(match.linkUrl)
       return `<a href="${encodedLink
-      }" target="_blank" rel="noopener noreferrer">${match.linkHtml || encodedLink
+      }" target="&#95;blank" rel="noopener noreferrer">${match.linkHtml || encodedLink
```
[the remark this answers is not in the document]
Comment (achitJ) [reply src/index.js:496]: THis is fine since we are using this internally only

File: test/controlSequenceTest.js [hunk 4]
```diff
@@ -53,25 +53,25 @@ describe('control sequences', () => {
 
   describe('hyperlinks', () => {
     it('should render an anchor tag', () => {
-      escapeForSlack('<https://swiftype.com>').should.equal('<a href="https://swiftype.com" target="_blank" rel="noopener noreferrer">https://swiftype.com</a>')
+      escapeForSlack('<https://swiftype.com>').should.equal('<a href="https://swiftype.com" target="&#95;blank" rel="noopener noreferrer">https://swiftype.com</a>')
```
[the remark this answers is not in the document]
Comment (achitJ) [reply test/controlSequenceTest.js:56]: THis is fine since we are using this internally only

**Review by batflarrow — APPROVED** [review]
